### PR TITLE
workerapi: offer QueryNow in worker thrift

### PR DESCRIPTION
**Pull request by dbentley**
Expands the Worker Thrift API to also offer the QueryNow functionality from runner.Service.

### CODE REVIEW

**Comment by JeanetteBruno** [comment]
What is the design for the worker maintaining a history of task requests and their states?  Don't we have the saga implementation for this information.... I thought the worker api is synchronous, so it only has one active request at a time.  (I think I'm missing something.)

**Comment by dbentley** [comment]
1) The worker will maintain history only as long as it's alive, with the in-memory statuses impl before.
2) Yes, this information wants to be in the saga log. The worker thrift API is just to get the data from the worker to the scheduler.
3) The API is not synchronous. Thrift doesn't like RPC's that last the 40 minutes running on the worker can take.
4) Currently, yes, we only run one at a time, but there's no fundamental reason we couldn't be running multiple runs on one worker at a time.

File: workerapi/api_test.go [hunk 1]
```diff
@@ -137,6 +149,47 @@ var tests = []struct {
...
+		},
+		[]runner.RunStatus{
+			runner.RunStatus{RunID: "a", State: runner.COMPLETE,
+				StdoutRef: "", StderrRef: "", ExitCode: 0, Error: ""},
+			runner.RunStatus{RunID: "b", State: runner.COMPLETE,
+				StdoutRef: "", StderrRef: "", ExitCode: 0, Error: ""},
+		},
+	},
+
+	// RunQuery
```
Comment (dgassaway) [inline workerapi/api_test.go]: Looks like thrift struct RunsQuery fields are all optional - worth adding a test case with empty struct to make sure that works as intended (does nothing?)

File: workerapi/client/client.go [hunk 2]
```diff
@@ -134,12 +133,19 @@ func (c *simpleClient) StatusAll() ([]runner.RunStatus, error) {
 	return st.Runs, nil
 }
 
+// Implements Scoot Worker API
 func (c *simpleClient) QueryNow(q runner.Query) ([]runner.RunStatus, error) {
```
Comment (JeanetteBruno) [inline workerapi/client/client.go]: While we are in here, can we change the name to be consistent with our other run status retrieval apis (right now we have Status, StatusAll and QueryNow).... can we change QueryNow to FilteredStatus (or something with the word Status in it)

**Comment by dgassaway** [comment]
@dbentley I think this was superseded a while back?
